### DISCUSSION TITLE
测试和生产在同一个Apollo Portal管理下，怎么升级Apollo比较好？

### DISCUSSION THREAD

**Discussion by kyleli666** [apolloconfig/apollo#5205] [Q&A] [answered]
我们想升级Apollo从2.1.0到2.2.0、2.3.0等，并且想在测试环境先升级看一下情况，没问题后再升级生产环境。
但是我们的测试和生产在同一个Apollo Portal管理下，请教一下升级步骤怎么安排比较好？

按照release note里写的升级顺序，是不是应该先只升级测试环境不动Portal，然后升级生产环境，最后升级Portal？还是说Portal可以先升级的？

**Comment by nobodyiam** [comment, marked as answer]
先升级 config service 和 admin service，最后升级 portal

**Reply by kyleli666** [reply]
就是说等到测试和生产环境的config 和 admin service 都升级完了，最后再升级portal是么？

**Reply by nobodyiam** [reply]
是的

**Reply by kyleli666** [reply]
好的~谢谢！
